src/test/java/jewellery/inventory/unit/service/SaleServiceTest.java

```diff
@@ -0,0 +1,111 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class SaleServiceTest {
+  @InjectMocks private SaleService saleService;
+  @Mock private SaleRepository saleRepository;
+  @Mock private UserService userService;
+  @Mock private ProductService productService;
+  @Mock private ProductRepository productRepository;
+  @Mock private SaleMapper saleMapper;
+  private User seller;
+  private User buyer;
+  private Product product;
+  private Sale sale;
+  private SaleRequestDto saleRequestDto;
+  private SaleRequestDto saleRequestDtoSellerNotOwner;
+  private SaleResponseDto saleResponseDto;
+  private ProductPriceDiscountRequestDto productPriceDiscountRequestDto;
+  private List<Product> productsForSale;
+
+  @BeforeEach
+  void setUp() {
+    seller = createTestUser();
+    seller.setId(UUID.randomUUID());
+    buyer = createSecondTestUser();
+    product = getTestProduct(seller, new Resource());
+    productsForSale = SaleTestHelper.getProductsList(product);
+    sale = SaleTestHelper.createSaleWithTodayDate(seller, buyer, productsForSale);
+    saleResponseDto = SaleTestHelper.getSaleResponseDto(sale);
+    productPriceDiscountRequestDto =
+        SaleTestHelper.createProductPriceDiscountRequest(product.getId(), 1000, 10);
+    List<ProductPriceDiscountRequestDto> productPriceDiscountRequestDtoList = new ArrayList<>();
+    productPriceDiscountRequestDtoList.add(productPriceDiscountRequestDto);
+    saleRequestDto =
+        SaleTestHelper.createSaleRequest(
+            seller.getId(), buyer.getId(), productPriceDiscountRequestDtoList);
+    saleRequestDtoSellerNotOwner =
+        SaleTestHelper.createSaleRequest(
+            buyer.getId(), buyer.getId(), productPriceDiscountRequestDtoList);
+  }
+
+  @Test
+  void testGetAllSales() {
+
+    List<Sale> sales = Arrays.asList(sale, new Sale(), new Sale());
+
+    when(saleRepository.findAll()).thenReturn(sales);
+
+    List<SaleResponseDto> responses = saleService.getAllSales();
+
+    Assertions.assertEquals(sales.size(), responses.size());
+  }
+
+  @Test
+  void testCreateSaleProductWillSuccessfully() {
+    when(saleMapper.mapRequestToEntity(saleRequestDto, seller, buyer, List.of(product)))
+        .thenReturn(sale);
+    when(userService.getUser(any(UUID.class))).thenReturn(seller, buyer);
+    when(saleRepository.save(sale)).thenReturn(sale);
+    when(productService.getProduct(any(UUID.class))).thenReturn(product);
+
+    when(saleMapper.mapEntityToResponseDto(sale)).thenReturn(saleResponseDto);
+
+    SaleResponseDto actual = saleService.createSale(saleRequestDto);
+    assertNotEquals(actual.getBuyer(), actual.getSeller());
+    Assertions.assertEquals(saleRequestDto.getSellerId(), actual.getSeller().getId());
+    assertNotNull(actual);
```

Comment:
You should either move this assertion above or remove it completely. Right now because of the previous assertions, if actual is null, it would throw a nullpointer before reaching this line.